src/hosts/host.entity.mjs

```diff
@@ -46,8 +47,9 @@ export class HostEntity {
   */
   constructor (options) {
     options && Object.assign(this, options)
-
-    this.oid ??= nanoid()
-    this.pid ??= nanoid(128)
+    const customAlphabetOid = nanoid.customAlphabet(config.oid.charset, +config.oid.length)
```

Comment:
The config is expected to be immutable during the application run, so these two can be moved to the top as two methods, for example: 

```js
const generateOID = nanoid.customAlphabet(config.oid.charset, +config.oid.length)
const generatePID = nanoid.customAlphabet(config.pid.charset, +config.pid.length)
```